spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -230,9 +243,11 @@ else if (configs.get("spring.json.default.value.type") instanceof String) {
 			else if (!isKey && configs.containsKey(VALUE_DEFAULT_TYPE)) {
 				if (configs.get(VALUE_DEFAULT_TYPE) instanceof Class) {
 					this.targetType = (Class<T>) configs.get(VALUE_DEFAULT_TYPE);
+					this.reader = this.objectMapper.readerFor(this.targetType);
 				}
 				else if (configs.get(VALUE_DEFAULT_TYPE) instanceof String) {
 					this.targetType = (Class<T>) ClassUtils.forName((String) configs.get(VALUE_DEFAULT_TYPE), null);
+					this.reader = this.objectMapper.readerFor(this.targetType);
```

Comment:
Hm. Can't all these `readerFor()` calls go to the end of this method, e.g. before `addTargetPackageToTrusted();` ?